### PR TITLE
:seedling: reorg the topics

### PR DESCRIPTION
**Pull request by skeeey**
## Summary

reorg the topics to:

- SourceEvents topic (`sources/+/clusters/+/sourceevents`) is a topic for sources to publish their resource create/update/delete events or status resync events
- AgentEvents topic (`sources/+/clusters/+/agentevents`) is a topic for agents to publish their resource status update events or spec resync events
- (Optional) SourceBroadcast (`sources/+/sourceevents`)  is a topic for a source to publish its events to all agents
- (Optional) AgentBroadcast (`clusters/+/agentevents`) is a topic for a agent to publish its events to all sources

### CODE REVIEW

**Comment by clyang82** [comment]
Could you append the description about how many topics we have for status and the responsible for each status?

**Comment by skeeey** [comment]
> Could you append the description about how many topics we have for status and the responsible for each status?

added, more details can be found the  comments on the `Topics` structrue

File: pkg/cloudevents/generic/interface.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ type Codec[T ResourceObject] interface {
 
 type CloudEventsClient[T ResourceObject] interface {
 	// Resync the resources of one source/agent by sending resync request.
-	Resync(ctx context.Context, listOptions types.ListOptions) error
+	Resync(context.Context, string) error
```
Comment (qiujian16) [inline pkg/cloudevents/generic/interface.go:57]: comment on what is the second variable

File: pkg/cloudevents/generic/types/types.go [hunk 2]
```diff
@@ -82,16 +79,47 @@ const (
...
+	//     resync request
+	// The topic format is `^([a-z]+)/([a-z0-9-]+|\+)/([a-z]+)/\+/agentevents$`, e.g. sources/+/clusters/+/agentevents
+	// or sources/source1/clusters/+/agentevents
+	AgentEvents string `json:"agentEvents" yaml:"agentEvents"`
+
+	// SourceBroadcast is an optional topic, it is for a source to publish its events to all agents, currently, we use
+	// this topic to resync resource status from all agents for a source that does not known the exact agents, e.g.
+	//   - A source uses this topic to publish its resource status resync request with its sourceID to all the agents
+	//   - Each agent subscribes to this topic to response sources resource status resync request
+	// The topic format is `^([a-z]+)/([a-z0-9-]+|\+)/sourceevents$`, e.g. sources/source1/sourceevents or
```
Comment (qiujian16) [inline pkg/cloudevents/generic/types/types.go:113]: maybe agentbroadcast/sourcebroadcast So it is different from agentevents/sourceevents

File: pkg/cloudevents/generic/interface.go (hunk 1 above)
Comment (skeeey) [reply pkg/cloudevents/generic/interface.go:62]: added

File: pkg/cloudevents/generic/types/types.go (hunk 2 above)
Comment (skeeey) [reply pkg/cloudevents/generic/types/types.go]: updated

**Comment by skeeey** [comment]
we may also need enable this option in the https://github.com/open-cluster-management-io/sdk-go/blob/main/pkg/cloudevents/work/clientbuilder.go#L58

**Pull request merged**